sql/log_event.cc

```diff
@@ -6984,10 +6984,12 @@ int Rotate_log_event::do_update_pos(Relay_log_info *rli)
     thd->backup_binlog_lock.release_protection(thd);
 
     if (rli->is_parallel_exec())
+    {
+      time_t ts= when.tv_sec + static_cast<time_t>(exec_time);
```

Comment:
Do not initialize ts if !server_id?
I am not sure how rli->is_parallel_exec() and !server_id can be true at the same time, but this is outside the scope